src/hooks/useKyberSwapConfig.ts

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
+
+import { ChainId, SUPPORTED_NETWORKS } from 'constants/networks'
+import { useLazyGetKyberswapConfigurationQuery, KyberswapConfig, parseResponse } from '../services/ksSetting'
+import { AppState } from 'state'
+
+export const useKyberswapConfig = (): {
+  [chain in ChainId]: KyberswapConfig
+} => {
+  const storeChainIds = useSelector<AppState, ChainId[]>((state) => state.application.activeNetworksId) // read directly from store instead of useActiveWeb3React to prevent circular loop
+  const [kyberswapConfigs, setKyberswapConfigs] = useState<
+    | {
+        [chain in ChainId]: KyberswapConfig
+      }
+    | null
+  >(null)
+  const [getKyberswapConfiguration, a] = useLazyGetKyberswapConfigurationQuery()
+
+  useEffect(() => {
+    const run = async () => {
+      setKyberswapConfigs(null)
+      const fetches = storeChainIds.map(async (chainId) => {
+        try {
+          const { data, error } = await getKyberswapConfiguration({ chainId })
+          if (data)
+            return {
+              chainId,
+              result: data,
+            }
+          if (error)
```

Comment:
remove if error

Reply:
I want to leave it, for any change in logic later would be easily.

Reply:
![image](https://user-images.githubusercontent.com/33005392/231675972-230cf153-a869-46a2-aa2d-79d048d927e3.png)
same result. no need if error

Reply:
I know what u r mentioning to. I prefer to leave it as is.

Reply:
:dungz